common/overlaysynccleanuphandler.h

```diff
@@ -43,6 +43,17 @@ class OverlaySyncCleanupHandler : public CleanUpHandler
      * @return ReturnCode::FAILURE on failure
      */
     ReturnCode clean() override;
+
+    /**
+     * @brief queryName is needed to query member name yet its full functionality is not needed for
+     * this class.
+     *
+     * Since OverlaySyncCleanupHandler uses DirectoryCleanUpHandler for removing directory, this
+     * function won't be called for the same reason.
+     *
+     * @return an empty string
+     */
+    const std::string queryName() override;
```

Comment:
Sorry, but I still don't like this description. OverlaySyncCleanupHandler doesn't use DirectoryCleanupHandler, it uses RecursiveCopy.

The reason why we're returning an empty string here is because we don't want any paths to compare equal to an overlay path, isn't it? Then we should write that.